test/integration/client/client_test.go

```diff
@@ -1347,3 +1359,494 @@ func TestExtractModifyApply_ForceOwnership(t *testing.T) {
+	// ensure that the CBOR client feature gates do not interfere with this.
+	DoRequestWithProtobufPreferredGeneratedClient := func(t *testing.T, config *rest.Config) error {
+		clientset, err := clientset.NewForConfig(config)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		_, err = clientset.CoreV1().Namespaces().Create(
+			context.TODO(),
+			&v1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test-generated-client-cbor-enablement",
+				},
+			},
+			metav1.CreateOptions{DryRun: []string{metav1.DryRunAll}},
+		)
+		return err
+	}
+
+	DoRequestWithGeneratedClient := func(t *testing.T, config *rest.Config) error {
+		// This is using a generated client from sample-apiserver because it is generated
+		// without --prefer-protobuf. For convenience, the test serves the API as a CRD with
+		// a permissive schema instead of running a real aggregated sample-apiserver.
+		wardleClient, err := wardlev1alpha1client.NewForConfig(config)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		_, err = wardleClient.Fischers().Create(
+			context.TODO(),
+			&wardlev1alpha1.Fischer{
+				ObjectMeta: metav1.ObjectMeta{Name: "test-generated-client-cbor-enablement"},
+			},
+			metav1.CreateOptions{DryRun: []string{metav1.DryRunAll}},
+		)
+		return err
+	}
+
+	type testCase struct {
+		name                    string
+		served                  bool
+		allowed                 bool
+		preferred               bool
+		configuredContentType   string
+		configuredAccept        string
+		wantRequestContentType  string
+		wantRequestAccept       string
+		wantResponseContentType string
+		wantResponseStatus      int
+		wantStatusError         bool
+		doRequest               func(t *testing.T, config *rest.Config) error
+	}
+
+	testCases := []testCase{
+		{
+			name:                    "cbor allowed and preferred client forces protobuf",
+			served:                  true,
+			allowed:                 true,
+			preferred:               true,
+			wantRequestContentType:  "application/vnd.kubernetes.protobuf",
+			wantRequestAccept:       "application/vnd.kubernetes.protobuf,application/json",
+			wantResponseContentType: "application/vnd.kubernetes.protobuf",
+			wantResponseStatus:      http.StatusCreated,
+			wantStatusError:         false,
+			doRequest:               DoRequestWithProtobufPreferredGeneratedClient,
+		},
+		{
+			name:                    "cbor allowed and not preferred client forces protobuf",
+			served:                  true,
+			allowed:                 true,
+			preferred:               false,
+			wantRequestContentType:  "application/vnd.kubernetes.protobuf",
+			wantRequestAccept:       "application/vnd.kubernetes.protobuf,application/json",
+			wantResponseContentType: "application/vnd.kubernetes.protobuf",
+			wantResponseStatus:      http.StatusCreated,
+			wantStatusError:         false,
+			doRequest:               DoRequestWithProtobufPreferredGeneratedClient,
```

Comment:
why does a test labelled "not preferred client forces protobuf" use a client called `DoRequestWithProtobufPreferredGeneratedClient`?

Reply:
The "client prefers CBOR" gate that controls the default request encoding is disabled.

Reply:
Thanks.